pandas/core/dtypes/common.py

```diff
@@ -152,8 +153,22 @@ def is_sparse(arr):
     >>> is_sparse(bsr_matrix([1, 2, 3]))
     False
     """
+    from pandas.core.sparse.array import SparseArray
+    from pandas.core.sparse.dtype import SparseDtype
+    from pandas.core.generic import ABCSeries
+    from pandas.core.internals import BlockManager, Block
 
-    return isinstance(arr, (ABCSparseArray, ABCSparseSeries))
+    if isinstance(arr, BlockManager):
+        # SparseArrays are only 1d
+        if arr.ndim == 1:
+            arr = arr.blocks[0]
+        else:
+            return False
+
+    if isinstance(arr, (ABCSeries, Block)):
```

Comment:
can you use ``getattr(arr, 'values', arr)``?

Reply:
Don't think so, since that would densify a `SparseArray`, and then return False.